test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java

```diff
@@ -0,0 +1,56 @@
+package org.scenarioo.pizza.scenarioo;
+
+import org.scenarioo.api.ScenarioDocuWriter;
+import org.scenarioo.model.docu.entities.Branch;
+import org.scenarioo.model.docu.entities.Build;
+
+import java.io.File;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Date;
+
+
+public class ScenariooBuildContext {
+
+    private static final String BRANCH_NAME = "pizza-delivery-" + envOrDefault("BRANCH_NAME", "master");
+    private static final String BUILD_NAME = envOrDefault("BUILD_NUMBER", "undefined");
+    private static final File DOCUMENTATION_ROOT = new File("build/scenariooDocumentation");
+
+    static {
+        DOCUMENTATION_ROOT.mkdirs();    // Create folder for generated documentation
+        ScenarioDocuWriter writer = getNewWriter();
+        writer.saveBranchDescription(createBranch(BRANCH_NAME));    // Create 'branch.xml'
+        writer.saveBuildDescription(createBuild(BUILD_NAME));       // Create 'build.xml'
+        writer.flush();
+    }
+
```

Comment:
I would recommend to change this, as follows:

```
static {
           // static initialization to ensure the build directory is prepared 
           // and build and branch description is written only once for all tests
           // (you could do this as well at the end of your build scripts).
           initScenariooReportForCurrentBuildRun();
 }

/**
 * Create branch and build directory for current scenarioo report build being generated
 * including the needed description files for the branch and the build.
 * This has to be done at least once for a full scenarioo report build.
 */
private void initScenariooReportForCurrentBuildRun() {
           DOCUMENTATION_ROOT.mkdirs();
           ScenarioDocuWriter writer = getNewWriter();
           writer.saveBranchDescription(createBranch(BRANCH_NAME));
           writer.saveBuildDescription(createBuild(BUILD_NAME));
           writer.flush();
     }
```
Reasons:
 1. better explanations
 2. I do not like inline comments behind code, but rather good method names and some good explanations of whole blocks/methods instead.
 3. I think `// Create branch.xml` is too technical and not realy important (why should the user care that this creates a XML ? could as well be a JSON later. the name of the method allready says all that he needs to know kind of).

Reply:
@ideco I changed my proposal above again  -what do you think? Makes sense like that?

Reply:
Absolutely!